chart/openfaas/templates/prometheus-external-svc.yaml

```diff
@@ -1,25 +0,0 @@
-{{- if .Values.exposeServices }}
----
-apiVersion: v1
-kind: Service
-metadata:
-  labels:
-    app: {{ template "openfaas.name" . }}
-    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
-    component: prometheus
-    heritage: {{ .Release.Service }}
-    release: {{ .Release.Name }}
-  name: prometheus-external
-  namespace: {{ .Release.Namespace | quote }}
-spec:
-  type: {{ .Values.serviceType }}
```

Comment:
I think this should remain in the chart - if someone requests `"NodePort" .Values.serviceType` then the NodePort for Prometheus should be exposed. This is especially useful for development and bare-metal Kubernetes.

Reply:
What about production on baremetal? You would expose Prom to everyone. With a baremetal cloud like Scaleway where you have no firewall this is very unsafe. I think the Helm chart should be secure by default, if you want to expose Prometheus for `dev` you can do `kubectl port-forward -n openfaas deployment/prometheus 9090:9090`

Reply:
I agree, using `port-forward` is pretty easy. We could add a comment in the readme or in the `NOTES.txt` to document this?

Reply:
The problem is that this is a breaking change and you haven't updated any docs and there are many tutorials that talk about this NodePort.

Reply:
https://docs.openfaas.com/deployment/kubernetes/

Reply:
Just the top few I found on Google:

https://medium.com/@IrekRomaniuk/openfaas-with-kubernetes-194e8eec48ab

http://panosgeorgiadis.com/blog/2017/11/08/how-to-start-with-openfaas/

https://github.com/justindavies/ServerlessOnAKS

https://blogs.technet.microsoft.com/livedevopsinjapan/2017/09/19/open-faas-on-acs-kubernetes/

https://hackernoon.com/deploying-kubernetes-on-premise-with-rke-and-deploying-openfaas-on-it-part-2-cc14004e7007